src/core/lib/support/reference_counted.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/lib/support/reference_counted.h"
+
+#include <grpc/support/log.h>
+
+#include "src/core/lib/support/memory.h"
+
+namespace grpc_core {
+
+void ReferenceCounted::Ref(const DebugLocation& location, const char* reason) {
```

Comment:
Move implementation to header (it makes sense to pick up automatic inlining for this function)